src/Workspaces/Core/Portable/Workspace/Solution/Solution.cs

```diff
@@ -548,7 +613,21 @@ public Solution AddMetadataReference(ProjectId projectId, MetadataReference meta
         /// </summary>
         public Solution AddMetadataReferences(ProjectId projectId, IEnumerable<MetadataReference> metadataReferences)
         {
-            var newState = _state.AddMetadataReferences(projectId, metadataReferences);
+            CheckContainsProject(projectId);
+
+            // avoid enumerating multiple times:
+            var collection = metadataReferences?.ToCollection();
```

Comment:
Do you want to reassign back to "metadataReferences" to prevent somebody from accidentally touching it again? (Question applies to rest of pattern in this file.)